providers/amt/amt.go

```diff
@@ -0,0 +1,178 @@
+package amt
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/ammmze/go-amt"
+	"github.com/bmc-toolbox/bmclib/providers"
+	"github.com/go-logr/logr"
+	"github.com/jacobweinstock/registrar"
+)
+
+const (
+	// ProviderName for the provider AMT implementation
+	ProviderName = "AMT"
```

Comment:
Mind if we call this package and its `ProviderName` 'IntelAMT' instead, just so its a bit more indicative of the hardware vendor in the provider name.

Reply:
sure thing